zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -216,9 +217,19 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
     String superUser = X509AuthenticationConfig.getInstance().getZnodeGroupAclSuperUserId();
 
     Set<Id> newAuthIds = new HashSet<>();
+
+    // Find interesecting super user domains/cross domains from provided domains list
+    Set<String> commonSuperUserDomains =
+        superUserDomainNames.stream().filter(domains::contains).collect(Collectors.toSet());
+
     // Check if user belongs to super user group
     if (clientId.equals(superUser)) {
       newAuthIds.add(new Id(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME, clientId));
+    } else if (!commonSuperUserDomains.isEmpty()) {
+      // For cross domain components, add (super:domainName) in authInfo
+      // "super" scheme gives access to all znodes without checking znode ACL vs authorized domain name
+      commonSuperUserDomains.stream().forEach(d ->
+          newAuthIds.add(new Id(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME, d)));
```

Comment:
Good point raised! so i think about below points : 
1. We don't expect that dedicated domain will also be present in cross domains list so it should not affect this. Above line will add "super" scheme for all the domains present in "cross domain list". I verified that it works and creates znode with world,all acls
2. Even if dedicated domain is listed in cross domain lists, cross domain will be prioritized and get "super" as scheme. So again this would be same as case one and it won't set any ACLs on any of the znodes created.

Reply:
Discussed offline. For dedicated domain component cases, the current code in `PrepRequestProcessor` won't set ACL to nodes, so wether setting domainName or clientId won't make a difference since no ACL will be set. Resolving this comment.